src/include/simeng/arch/riscv/Instruction.hh

```diff
@@ -47,7 +48,9 @@ enum class InstructionException {
   SupervisorCall,
   HypervisorCall,
   SecureMonitorCall,
-  NoAvailablePort
+  NoAvailablePort,
+  IllegalInstruction,
+  AtomicOperation
```

Comment:
@jj16791 @JosephMoore25

Reply:
I agree it probably shouldn't be `ProcessStateUpdate`, given we already have `ProcessStateChange`, and that it should also change. I'm not sure what cases this exception gets generated in. If it is due to a process state change, then even `ProcessStateFlush` could be a little more descriptive.  That being said, this could imply you're clearing the process state. Further thought required...

Reply:
The original intent was specifically for atomic operations (csr instructions, like this one, are meant to execute atomically) which may require some more specific logic later down the line regarding the ```NOTE```/```TODO`` at the top of the conditional body. I'm fine changing it now but I reckon it will likely change again in the near future when we have a clearer idea of how this will actually work

Reply:
Thumbs up for ```PipelineFlush```, thumbs down for ```AtomicOperation``` will change to whatever has the majority vote